### ISSUE TITLE
Unable to resolve external type: SEL

### ISSUE THREAD

**Issue by darth-cheney** [estebanlm/objcbridge#1]
Yo,

I am just testing out this library to see if we can use it to make classes for various Cocoa frameworks.
  
I tried the following code:
```smalltalk
nsAlert := ObjCClass lookup: 'NSAlert' inFramework: 'AppKit'.
nsAlert descriptionString. "doIt"
```
  
I get `Error: Unable to resolve external type: SEL`
  
Am I doing something wrong?

**Comment by estebanlm** [comment]
can you check executing 

```Smalltalk
ObjCLibrary initialize
```

will that work?

**Comment by darth-cheney** [comment]
Yes that works no problem

**Comment by estebanlm** [comment]
I mean, your example works after that?

**Comment by darth-cheney** [comment]
No it still doesn't work.
  
I've trotted through the stack in the debugger and for whatever reason all of the indirection that happens (with traits I guess) fails to resolve the selector mapping of `SEL` to `ObjCSelector` as present in the `ObjCTypes`. I can't determine more because it's hard to follow what's going on.

**Comment by estebanlm** [comment]
Mmm.
You just spotted a bug in UFFI I thought it was corrected. 
The method `FFICallout>>resolveType:` should look like this: 

```Smalltalk
resolveType: aTypeName
	" a type name could be
	 - a class variable name
	- a class name
	- a type name 
	- a type name, followed by arbitrary number pointer chars - $*"
	| name newName resolver binding ptrArity |
	
	newName := aTypeName.
	ptrArity := 0.
	"resolve aliases and pointers"	
	[
		name := newName asString trimRight.  
		newName := self aliasForType: name.
		newName last = $* ifTrue: [
			ptrArity := ptrArity + 1.
			newName := newName allButLast ].
		name = newName ] whileFalse.
	
	resolver := requestor 
		ifNotNil: [ 
			requestor isClass
				ifTrue: [ requestor theNonMetaClass ]
				ifFalse: [ requestor ] ]
		ifNil: [ self class ].
	binding := resolver ffiBindingOf: name asSymbol.
	
	binding ifNotNil: [ 
		^ (binding value asExternalTypeOn: self) pointerArity: ptrArity ] .

	^ self error: 'Unable to resolve external type: ', aTypeName
```

And is not. Please change it and test it. If everything works for you I will submit the fix.

**Comment by darth-cheney** [comment]
Yup, that seems to have done the trick!

**Comment by estebanlm** [comment]
I sent a fix for Pharo to solve this problem : https://github.com/pharo-project/pharo/pull/1894
